base/replicator.go

```diff
@@ -38,22 +38,18 @@ func NewReplicator() *Replicator {
 }
 
 func (r *Replicator) Replicate(params sgreplicate.ReplicationParameters, isCancel bool) (task *ActiveTask, err error) {
+
+	replicationId, found := r.getReplicationForParams(params)
+
 	if isCancel {
-		replicationId := params.ReplicationId
 		// If replicationId isn't defined in the cancel request, attempt to look up the replication based on source, target
 		if replicationId == "" {
```

Comment:
Removed the redundant if replicationId == "", also noticed cancel was returning 'null', it now returns the active task state after stopping the replication.